src/neuroglancer/datasource/boss/api.ts

```diff
@@ -27,14 +27,26 @@ export type BossToken = string;
 export const credentialsKey = 'boss';
 
 export function fetchWithBossCredentials<T>(
-    credentialsProvider: CredentialsProvider<BossToken>, input: RequestInfo, init: RequestInit,
-    transformResponse: ResponseTransform<T>,
-    cancellationToken: CancellationToken = uncancelableToken): Promise<T> {
-  return fetchWithCredentials(
+  credentialsProvider: CredentialsProvider<BossToken>, input: RequestInfo, init: RequestInit,
+  transformResponse: ResponseTransform<T>,
+  cancellationToken: CancellationToken = uncancelableToken): Promise<T> {
+  return cancellableFetchOk(input, init, transformResponse, cancellationToken).catch((error) => {
+    if (error.status !== 500 && error.status !== 401 && error.status !== 403 && error.status !== 504) {
+      // Prevent an infinite loop of error = 0 where the request
+      // has been cancelled
+      throw error;
+    }
+    return fetchWithCredentials(
       credentialsProvider, input, init, transformResponse,
       credentials => {
         const headers = new Headers(init.headers);
-        headers.set('Authorization', `Bearer ${credentials}`);
+        if (typeof(credentials) === "string") {
+          headers.set('Authorization', `Bearer ${credentials}`);
+        }
+        else {
+          //@ts-ignore
+          headers.set('Authorization', `${credentials.tokenType} ${credentials.accessToken}`); 
```

Comment:
If you need to specify something other than Bearer sometimes, just change `BossToken` to `{tokenType: string, accessToken: string}` and always use that form, rather than allowing two different ways of specifying it.